Source/WebCore/page/EventHandler.cpp

```diff
@@ -4337,49 +4337,107 @@ void EventHandler::defaultBackspaceEventHandler(KeyboardEvent& event)
         event.setDefaultHandled();
 }
 
-KeyboardScrollingAnimator* EventHandler::keyboardScrollingAnimatorForFocusedNode()
+void EventHandler::stopKeyboardScrolling()
 {
-    Node* node = m_frame.document()->focusedElement();
+    auto animator = m_frame.page()->currentKeyboardScrollingAnimator();
+    if (animator)
+        animator->handleKeyUpEvent();
+}
 
-    if (!node)
-        node = m_mousePressNode.get();
+bool EventHandler::beginKeyboardScrollGesture(KeyboardScrollingAnimator* animator, KeyboardEvent& event)
+{
+    auto* platformEvent = event.underlyingPlatformEvent();
 
-    auto* scrollableArea = enclosingScrollableArea(node);
+    if (animator && platformEvent && animator->beginKeyboardScrollGesture(*platformEvent)) {
+        m_frame.page()->setCurrentKeyboardScrollingAnimator(animator);
+        return true;
+    }
+
+    return false;
+}
+
+bool EventHandler::startKeyboardScrollAnimationOnDocument(KeyboardEvent& event)
+{
+    RefPtr frame = &m_frame;
```

Comment:
I don't think we need RefPtrs here since nothing called from this function can invalidate the frame or view, and there's only one line that uses the view.